src/CachePool.php

```diff
@@ -40,13 +40,18 @@ class CachePool implements CacheInterface
      * @var mixed
      */
     protected $defaultValue;
+    /**
+     * @var int|DateInterval|null
+     */
+    protected $defaultTtl;
 
     /**
      * @param wpdb   $wpdb         The WP database object.
      * @param string $poolName     The name of this cache pool. Must be unique to this instance.
      * @param mixed $defaultValue  A random value. Used for false-negative detection. The more chaotic - the better.
+     * @param int|DateInterval|null $defaultTtl Optional default TTL to use when caching new entries.
      */
-    public function __construct(wpdb $wpdb, string $poolName, $defaultValue)
+    public function __construct(wpdb $wpdb, string $poolName, $defaultValue, $defaultTtl = null)
```

Comment:
What about making this explicitly `0`? Hopefully this would give instantiators a better idea of what the default will be when consumers use `set()`.

Reply:
Hmm ... I used `null` to mimic the original fallback behavior in `set()`. But it _would_ make more sense to replace that fallback with the default value.

Alright. Changing it

Reply:
Another approach could be to make the parameter nullable yet required.

Reply:
I'd prefer a zero default. It would be annoying to have to pass a required `null` every time in the constructor.